### ISSUE TITLE
Stash line chunks of a altered file

### ISSUE THREAD

**Issue by dnsampaio** [eamodio/vscode-gitlens#674]
Hi,
would it be possible to add a stash change button over the inline-changes view? Just as stage change, I would like to stash single changes in a file. For example, if in one file we have changes in lines [0 - 3], [20-30], [90-100]. I would like to select to stash just the [20-30] chunk.

Regards
Diogo

**Comment by eamodio** [comment]
@dnsampaio do you mean _stage_ rather than _stash_?

**Comment by VulcanRav** [comment]
@eamodio I think stashing specific ranges, `git stash -p`.

I'm in need of a similar utility myself, would you consider adding such a functionality to gitlens?

**Comment by eamodio** [comment]
This isn't something that I will likely add myself, but if the community wants to provide a PR I'm certainly open to it -- depending on the complexity and scope of the changes required.

**Comment by dnsampaio** [comment]
> @dnsampaio do you mean _stage_ rather than _stash_?

No, I really want to stash partial changes of a file. Such functionality does not exist directly in git. What I need to do is to stage all other changes I don't want stashed, commit, stash the file, reset the commit.

**Comment by grass29** [comment]
This would be very useful
